libcontainerd/client_linux.go

```diff
@@ -400,3 +400,50 @@ func (en *exitNotifier) close() {
 func (en *exitNotifier) wait() <-chan struct{} {
 	return en.c
 }
+
+func (clnt *client) CreateCheckpoint(containerID string, checkpointID string, exit bool) error {
+	clnt.lock(containerID)
+	defer clnt.unlock(containerID)
+	if _, err := clnt.getContainer(containerID); err != nil {
+		return err
+	}
+	_, err := clnt.remote.apiClient.CreateCheckpoint(context.Background(), &containerd.CreateCheckpointRequest{
+		Id: containerID,
+		Checkpoint: &containerd.Checkpoint{
+			Name:        checkpointID,
+			Exit:        exit,
+			Tcp:         true,
+			UnixSockets: true,
+			Shell:       true,
```

Comment:
This Shell: true makes me a little bit nervous. Does container share hist sid or terminal with someone else?